Can't not find bus

간헐적으로 아래 에러 발생

```java
Caused by: java.lang.IllegalArgumentException: Can't not find bus: 228000393
	at com.dku.council.infra.bus.model.mapper.BusResponseMapper.to(BusResponseMapper.java:15) ~[main/:na]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[na:na]
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[na:na]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) ~[na:na]
	at com.dku.council.infra.bus.provider.GGBusProvider.retrieveBusArrival(GGBusProvider.java:58) ~[main/:na]
	... 15 common frames omitted
```

--- a/src/main/java/com/dku/council/domain/comment/CommentRepository.java
+++ b/src/main/java/com/dku/council/domain/comment/CommentRepository.java
@@ -4,11 +4,20 @@ import com.dku.council.domain.comment.model.entity.Comment;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 import java.util.List;
 
 public interface CommentRepository extends JpaRepository<Comment, Long> {
+
+    @Query("select c from Comment c " +
+            "where c.post.id=:postId " +
+            "and (c.status='ACTIVE' or c.status='EDITED')")
     Page<Comment> findAllByPostId(Long postId, Pageable pageable);
 
+    @Query("select c from Comment c " +
+            "where c.post.id=:postId " +
+            "and c.user.id=:userId " +
+            "and (c.status='ACTIVE' or c.status='EDITED')")
     List<Comment> findAllByPostIdAndUserId(Long postId, Long userId);
 }

--- a/src/main/java/com/dku/council/domain/comment/service/CommentService.java
+++ b/src/main/java/com/dku/council/domain/comment/service/CommentService.java
@@ -98,6 +98,7 @@ public class CommentService {
         if (!comment.getUser().getId().equals(userId)) {
             throw new NotGrantedException();
         }
+
         CommentLog commentLog = CommentLog.builder()
                 .post(post)
                 .user(user)

--- a/src/main/java/com/dku/council/domain/post/repository/GenericPostRepository.java
+++ b/src/main/java/com/dku/council/domain/post/repository/GenericPostRepository.java
@@ -7,11 +7,11 @@ import org.springframework.data.jpa.repository.Query;
 
 import java.util.Optional;
 
-// TODO 단건조회시 한방 쿼리로 가져오기
 public interface GenericPostRepository<T extends Post> extends JpaRepository<T, Long>, JpaSpecificationExecutor<T> {
     /**
      * ACTIVE상태인 post만 가져옵니다.
      */
+    @Override
     @Query("select p from Post p where p.id=:id and p.status='ACTIVE'")
     Optional<T> findById(Long id);
 }

--- a/src/main/java/com/dku/council/infra/bus/model/mapper/BusResponseMapper.java
+++ b/src/main/java/com/dku/council/infra/bus/model/mapper/BusResponseMapper.java
@@ -6,13 +6,12 @@ import com.dku.council.infra.bus.model.BusStatus;
 import com.dku.council.infra.bus.model.ResponseGGBusArrival;
 import com.dku.council.infra.bus.model.ResponseKakaoBusApi;
 
-// todo Mapper Library를 사용해보는건 어떨까?
 public class BusResponseMapper {
     public static BusArrival to(ResponseGGBusArrival.Body.BusArrival model) {
         String routeId = model.getRouteId();
         Bus bus = Bus.of(routeId);
         if (bus == null) {
-            throw new IllegalArgumentException("Can't not find bus: " + routeId);
+            return null;
         }
 
         BusStatus state = getGGBusStateLabel(model.getFlag());

--- a/src/main/java/com/dku/council/infra/bus/provider/GGBusProvider.java
+++ b/src/main/java/com/dku/council/infra/bus/provider/GGBusProvider.java
@@ -15,6 +15,7 @@ import org.springframework.web.util.UriComponentsBuilder;
 
 import java.net.URI;
 import java.util.List;
+import java.util.Objects;
 import java.util.stream.Collectors;
 
 @Component
@@ -55,6 +56,7 @@ public class GGBusProvider implements BusArrivalProvider {
 
             return response.getMsgBody().getBusArrivalList().stream()
                     .map(BusResponseMapper::to)
+                    .filter(Objects::nonNull)
                     .collect(Collectors.toList());
         } catch (Throwable e) {
             throw new CannotGetBusArrivalException(e);

--- a/src/test/java/com/dku/council/domain/comment/CommentRepositoryTest.java
+++ b/src/test/java/com/dku/council/domain/comment/CommentRepositoryTest.java
@@ -1,0 +1,107 @@
+package com.dku.council.domain.comment;
+
+import com.dku.council.domain.comment.model.entity.Comment;
+import com.dku.council.domain.post.model.entity.posttype.GeneralForum;
+import com.dku.council.domain.post.repository.GeneralForumRepository;
+import com.dku.council.domain.user.model.entity.Major;
+import com.dku.council.domain.user.model.entity.User;
+import com.dku.council.domain.user.repository.MajorRepository;
+import com.dku.council.domain.user.repository.UserRepository;
+import com.dku.council.mock.CommentMock;
+import com.dku.council.mock.GeneralForumMock;
+import com.dku.council.mock.MajorMock;
+import com.dku.council.mock.UserMock;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@DataJpaTest
+class CommentRepositoryTest {
+
+    @Autowired
+    private MajorRepository majorRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private GeneralForumRepository generalForumRepository;
+
+    @Autowired
+    private CommentRepository repository;
+
+    private User user1;
+    private User user2;
+    private GeneralForum post1;
+    private GeneralForum post2;
+
+    @BeforeEach
+    void setUp() {
+        Major major = MajorMock.create();
+        major = majorRepository.save(major);
+
+        user1 = UserMock.create(major);
+        user1 = userRepository.save(user1);
+
+        user2 = UserMock.create(major);
+        user2 = userRepository.save(user2);
+
+        post1 = GeneralForumMock.create(user1);
+        post1 = generalForumRepository.save(post1);
+
+        post2 = GeneralForumMock.create(user1);
+        post2 = generalForumRepository.save(post2);
+
+        List<Comment> comments = CommentMock.createList(post1, List.of(user1, user2), 10);
+        comments.get(8).updateStatus(CommentStatus.DELETED);
+        comments.get(9).updateStatus(CommentStatus.DELETED);
+        repository.saveAll(comments);
+
+        List<Comment> comments2 = CommentMock.createList(post2, List.of(user1, user2), 7);
+        comments2.get(5).updateStatus(CommentStatus.DELETED);
+        comments2.get(6).updateStatus(CommentStatus.DELETED);
+        repository.saveAll(comments2);
+    }
+
+    @Test
+    void findAllByPostId() {
+        // when
+        Page<Comment> actual1 = repository.findAllByPostId(post1.getId(), Pageable.unpaged());
+        Page<Comment> actual2 = repository.findAllByPostId(post2.getId(), Pageable.unpaged());
+        Page<Comment> actual3 = repository.findAllByPostId(post1.getId(), Pageable.ofSize(5));
+        Page<Comment> actual4 = repository.findAllByPostId(post1.getId(), PageRequest.of(1, 5));
+
+        // then
+        assertThat(actual1.getTotalElements()).isEqualTo(8);
+        assertThat(actual2.getTotalElements()).isEqualTo(5);
+        assertThat(actual3.getTotalPages()).isEqualTo(2);
+        assertThat(actual3.getNumberOfElements()).isEqualTo(5);
+        assertThat(actual3.getNumber()).isEqualTo(0);
+        assertThat(actual3.getTotalElements()).isEqualTo(8);
+        assertThat(actual4.getNumberOfElements()).isEqualTo(3);
+        assertThat(actual4.getNumber()).isEqualTo(1);
+    }
+
+    @Test
+    void findAllByPostIdAndUserId() {
+        // when
+        List<Comment> actual1 = repository.findAllByPostIdAndUserId(post1.getId(), user1.getId());
+        List<Comment> actual2 = repository.findAllByPostIdAndUserId(post1.getId(), user2.getId());
+        List<Comment> actual3 = repository.findAllByPostIdAndUserId(post2.getId(), user1.getId());
+        List<Comment> actual4 = repository.findAllByPostIdAndUserId(post2.getId(), user2.getId());
+
+        // then
+        assertThat(actual1).hasSize(4);
+        assertThat(actual2).hasSize(4);
+        assertThat(actual3).hasSize(3);
+        assertThat(actual4).hasSize(2);
+    }
+}

--- a/src/test/java/com/dku/council/domain/comment/CommentRepositoryTest.java
+++ b/src/test/java/com/dku/council/domain/comment/CommentRepositoryTest.java
@@ -12,6 +12,7 @@ import com.dku.council.mock.GeneralForumMock;
 import com.dku.council.mock.MajorMock;
 import com.dku.council.mock.UserMock;
 import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
@@ -72,6 +73,7 @@ class CommentRepositoryTest {
     }
 
     @Test
+    @DisplayName("포스트 아이디로 댓글 조회 테스트")
     void findAllByPostId() {
         // when
         Page<Comment> actual1 = repository.findAllByPostId(post1.getId(), Pageable.unpaged());
@@ -91,6 +93,7 @@ class CommentRepositoryTest {
     }
 
     @Test
+    @DisplayName("포스트 아이디와 유저 아이디로 댓글 조회 테스트")
     void findAllByPostIdAndUserId() {
         // when
         List<Comment> actual1 = repository.findAllByPostIdAndUserId(post1.getId(), user1.getId());

--- a/src/test/java/com/dku/council/domain/comment/service/CommentServiceTest.java
+++ b/src/test/java/com/dku/council/domain/comment/service/CommentServiceTest.java
@@ -24,6 +24,8 @@ import java.util.Optional;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
@@ -73,11 +75,18 @@ class CommentServiceTest {
         when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
 
         // when
+        String prevComment = comment.getText();
         Long edited = service.edit(10L, 10L, "new text");
 
         // then
         assertThat(edited).isEqualTo(10L);
         assertThat(comment.getText()).isEqualTo("new text");
+        verify(commentLogRepository).save(argThat(log -> {
+            assertThat(log.getText()).isEqualTo(prevComment);
+            assertThat(log.getUser().getId()).isEqualTo(user.getId());
+            assertThat(log.getPost().getId()).isEqualTo(comment.getPost().getId());
+            return true;
+        }));
     }
 
     @Test

--- a/src/test/java/com/dku/council/infra/bus/provider/TownBusProviderTest.java
+++ b/src/test/java/com/dku/council/infra/bus/provider/TownBusProviderTest.java
@@ -37,7 +37,7 @@ class TownBusProviderTest extends AbstractMockServerTest {
         List<BusArrival> arrivals = service.retrieveBusArrival(BusStation.DKU_GATE);
 
         // then
-        assertThat(arrivals.size()).isEqualTo(2);
+        assertThat(arrivals.size()).isEqualTo(3);
         BusArrival arrival = arrivals.stream()
                 .filter(ent -> ent.getBusNo().equals("24"))
                 .findFirst()

--- a/src/test/java/com/dku/council/mock/CommentMock.java
+++ b/src/test/java/com/dku/council/mock/CommentMock.java
@@ -12,8 +12,9 @@ public class CommentMock {
 
     public static List<Comment> createList(Post post, List<User> user, int size) {
         List<Comment> comments = new ArrayList<>();
+        final int userSize = user.size();
         for (int i = 0; i < size; i++) {
-            comments.add(create(null, post, user.get(i)));
+            comments.add(create(null, post, user.get(i % userSize)));
         }
         return comments;
     }

--- a/src/test/java/com/dku/council/mock/GeneralForumMock.java
+++ b/src/test/java/com/dku/council/mock/GeneralForumMock.java
@@ -1,0 +1,26 @@
+package com.dku.council.mock;
+
+import com.dku.council.domain.post.model.entity.Post;
+import com.dku.council.domain.post.model.entity.posttype.GeneralForum;
+import com.dku.council.domain.user.model.entity.User;
+import com.dku.council.util.EntityUtil;
+
+public class GeneralForumMock {
+    public static GeneralForum create(User user) {
+        return create(user, null);
+    }
+
+    public static GeneralForum create(User user, Long id) {
+        GeneralForum post = GeneralForum.builder()
+                .user(user)
+                .title("title")
+                .body("body")
+                .build();
+
+        if (id != null) {
+            EntityUtil.injectId(Post.class, post, id);
+        }
+
+        return post;
+    }
+}

--- a/src/test/resources/mockdata/bus/getBusArrivalList.xml
+++ b/src/test/resources/mockdata/bus/getBusArrivalList.xml
@@ -24,6 +24,22 @@
             <stationId>228001978</stationId>
         </busArrivalList>
         <busArrivalList>
+            <flag>PASS</flag>
+            <locationNo1>1</locationNo1>
+            <locationNo2/>
+            <lowPlate1>0</lowPlate1>
+            <lowPlate2>0</lowPlate2>
+            <plateNo1>경기70아1122</plateNo1>
+            <plateNo2/>
+            <predictTime1>1</predictTime1>
+            <predictTime2/>
+            <remainSeatCnt1>32</remainSeatCnt1>
+            <remainSeatCnt2>0</remainSeatCnt2>
+            <routeId>200000000</routeId>
+            <staOrder>2</staOrder>
+            <stationId>228001978</stationId>
+        </busArrivalList>
+        <busArrivalList>
             <flag>WAIT</flag>
             <locationNo1>2</locationNo1>
             <locationNo2>5</locationNo2>

--- a/src/test/resources/mockdata/bus/kakao.json
+++ b/src/test/resources/mockdata/bus/kakao.json
@@ -119,6 +119,39 @@
         "collectStatus2": "NORMAL"
       },
       "running": false
+    },
+    {
+      "id": "B54367",
+      "name": "26-3",
+      "hcode": "31190",
+      "hname": "경기 용인시",
+      "type": "0109",
+      "busLineType": "MAUL",
+      "realTime": true,
+      "realtimeState": "NOVEHICLE",
+      "idx": "1/1",
+      "arrival": {
+        "direction": "대지고등학교.전내교차로 방향",
+        "busStopId": "BS264224",
+        "masterBusStopId": "BS83664",
+        "nextBusStopName": "대지고등학교.전내교차로",
+        "nextBusStopId": "BS264225",
+        "order": 18,
+        "vehicleType": "0",
+        "vehicleState": "1001",
+        "lastVehicle": false,
+        "arrivalTime": 0,
+        "busStopCount": 0,
+        "vehicleType2": "0",
+        "vehicleState2": "1001",
+        "lastVehicle2": false,
+        "arrivalTime2": 0,
+        "busStopCount2": 0,
+        "direction2": "",
+        "collectStatus": "NORMAL",
+        "collectStatus2": "NORMAL"
+      },
+      "running": true
     }
   ],
   "readableItsIdList": [

